### PR TITLE
Bump core to 5.6.36

### LINKED ISSUE #3432: Fix bug in FHIRPathEngine funcReplace
FHIRPathEngine funcReplace requires two parameters, but incorrectly throws an exception when two parameters are provided.

### PR DESCRIPTION
**Pull request by tadgh**
* Bring in fix for funcReplace exception in FHIRPathEngine. 
* Add changelog
* Add test proving replace works. 

Closes #3432

### CODE REVIEW

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3432-funcreplace-fix.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+type: fix
+issue: 3418
```
Comment (JasonRoberts-smile) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3432-funcreplace-fix.yaml:3]: This seems like the wrong issue number

File: hapi-fhir-structures-r5/src/main/java/org/hl7/fhir/r5/hapi/ctx/HapiWorkerContext.java [hunk 2]
```diff
@@ -281,8 +281,9 @@ public boolean hasCache() {
 	}
 
 	@Override
-	public ValueSetExpander.ValueSetExpansionOutcome expandVS(ConceptSetComponent theInc, boolean theHierarchical) throws TerminologyServiceException {
+	public ValueSetExpander.ValueSetExpansionOutcome expandVS(ConceptSetComponent theInc, boolean theHierarchical, boolean theNoInactive) throws TerminologyServiceException {
 		ValueSet input = new ValueSet();
+		input.getCompose().setInactive(!theNoInactive); //TODO GGG/DO is this valid?
```
Comment (JasonRoberts-smile) [inline hapi-fhir-structures-r5/src/main/java/org/hl7/fhir/r5/hapi/ctx/HapiWorkerContext.java:286]: I'm not so familiar with how we do value set expansion, but I dug around a bit and I couldn't find any obvious evidence of this inactive flag being used to filter whether inactive terms get included in the expansion, or used for anything in our code base for that matter. It seems to be beyond the scope of the current ticket, but worthy of follow-up in some form.

**Review by JasonRoberts-smile — APPROVED** [review]

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3432-funcreplace-fix.yaml (hunk 1 above)
Comment (tadgh) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3432-funcreplace-fix.yaml:3]: Right you are!

File: hapi-fhir-structures-r5/src/main/java/org/hl7/fhir/r5/hapi/ctx/HapiWorkerContext.java (hunk 2 above)
Comment (tadgh) [reply hapi-fhir-structures-r5/src/main/java/org/hl7/fhir/r5/hapi/ctx/HapiWorkerContext.java:286]: Yup, I am just following the implementation in the corelibs that used this. There was a breaking API change in core, so i bubbled up the base implementation here.

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3432-funcreplace-fix.yaml (hunk 1 above)
Comment (tadgh) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_0_0/3432-funcreplace-fix.yaml:3]: Corrected in 16f895ce42c554687f5b167c9ac82fb9fdff97fc

**Pull request merged**